array/array.go

```diff
@@ -0,0 +1,131 @@
+package array
+
+import (
+	"errors"
+)
+
+type Array struct {
+	arr  []int
+	size int
+}
+
+func NewArray(capacity int) *Array {
```

Comment:
We should have another constructor or give `capacity` a default value. 8 or 16 will be a good default capacity.